streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -377,9 +409,22 @@ public void setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler eh
     }
 
     private void validateIsRunning() {
-        if (state != RUNNING) {
+        if (state != KafkaStreamsState.RUNNING) {
             throw new IllegalStateException("KafkaStreams is not running");
         }
     }
 
+    /**
+     * @return The state this instance is in
+     */
+    public KafkaStreamsState getState() {
```

Comment:
it would be really useful to have the state exposed via a state listener interface that applications can register to take appropriate action on state change (e.g. similar to what curator does for connection state http://curator.apache.org/errors.html)